velox/type/tests/DecimalTest.cpp

```diff
@@ -176,5 +176,110 @@ TEST(DecimalTest, toByteArray) {
   testToByteArray(DecimalUtil::kLongDecimalMax, expected8, 16);
 }
 
+template <typename T>
+void checkRescaleDouble(double value, const TypePtr& type, T expectedValue) {
+  std::string error;
+  auto [precision, scale] = getDecimalPrecisionScale(*type);
+  std::optional<T> result =
+      DecimalUtil::rescaleDouble<T>(value, precision, scale, error);
+  ASSERT_TRUE(result.has_value());
+  ASSERT_TRUE(error.empty());
+  ASSERT_EQ(result.value(), expectedValue);
+}
+
+template <typename T>
+void checkRescaleDoubleFails(
+    double value,
+    const TypePtr& type,
+    const std::string& expectedError) {
+  std::string error;
+  auto [precision, scale] = getDecimalPrecisionScale(*type);
+  std::optional<T> result =
+      DecimalUtil::rescaleDouble<T>(value, precision, scale, error);
+  ASSERT_FALSE(result.has_value());
+  ASSERT_FALSE(error.empty());
+  ASSERT_EQ(error, expectedError);
+}
+
+TEST(DecimalTest, rescaleDouble) {
+  {
```

Comment:
these {} are not needed, are they?

Reply:
No, they are not needed. I added {} here to separate every group. I'll drop it.